presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1690,7 +1686,7 @@ private Scope analyzeFrom(QuerySpecification node, Scope scope)
         private void analyzeGroupingFunctions(QuerySpecification node, List<Expression> expressions)
         {
             boolean isGroupingOperationPresent = expressions.stream()
-                    .anyMatch(GroupingOperationAnalyzer::containsGroupingOperation);
+                    .anyMatch(expression -> !ExpressionTreeFilter.extract(groupingExtractionFunction, ImmutableList.of(expression)).isEmpty());
```

Comment:
Maybe add method to `ExpressionTreeFilter`:
`boolean hasExpressionsOfType(Expression expression, Class<? extends Expression> clazz)`?